apps/confidential/confidential.go

```diff
@@ -781,3 +787,45 @@ func (cca Client) Account(ctx context.Context, accountID string) (Account, error
 func (cca Client) RemoveAccount(ctx context.Context, account Account) error {
 	return cca.base.RemoveAccount(ctx, account)
 }
+
+// WithExtraBodyParameters adds extra body parameters to the token request.
+// These parameters are added to the request body and are also included in the cache key,
+// ensuring that tokens acquired with different parameters are cached separately.
+//
+// The parameter values are provided as functions that are evaluated at request time,
+// allowing for dynamic values that may change between requests.
+//
+// Example:
+//
+//	params := map[string]func(context.Context) (string, error){
+//	    "custom_param": func(ctx context.Context) (string, error) {
+//	        return "custom_value", nil
+//	    },
+//	}
+//	result, err := client.AcquireTokenByCredential(ctx, scopes, confidential.WithExtraBodyParameters(params))
+func WithExtraBodyParameters(params map[string]func(context.Context) (string, error)) interface {
+	AcquireByCredentialOption
+	AcquireSilentOption
+	options.CallOption
+} {
+
+	return struct {
+		AcquireByCredentialOption
+		AcquireSilentOption
+		options.CallOption
+	}{
+		CallOption: options.NewCallOption(
+			func(a any) error {
+				switch t := a.(type) {
+				case *acquireTokenByCredentialOptions:
+					t.extraBodyParameters = params
+				case *acquireTokenSilentOptions:
```

Comment:
nit: is it difficult to add it to the other confidential client flows? OBO and auth_code?

Reply:
Not difficult, i can add it.